examples/CSharp/Tutorial 3.2 - Diffuse Lighting/Program.cs

```diff
@@ -0,0 +1,226 @@
+             0.5f, -0.5f, -0.5f,  0.0f, -1.0f,  0.0f,
+             0.5f, -0.5f,  0.5f,  0.0f, -1.0f,  0.0f,
+             0.5f, -0.5f,  0.5f,  0.0f, -1.0f,  0.0f,
+            -0.5f, -0.5f,  0.5f,  0.0f, -1.0f,  0.0f,
+            -0.5f, -0.5f, -0.5f,  0.0f, -1.0f,  0.0f,
+
+            -0.5f,  0.5f, -0.5f,  0.0f,  1.0f,  0.0f,
+             0.5f,  0.5f, -0.5f,  0.0f,  1.0f,  0.0f,
+             0.5f,  0.5f,  0.5f,  0.0f,  1.0f,  0.0f,
+             0.5f,  0.5f,  0.5f,  0.0f,  1.0f,  0.0f,
+            -0.5f,  0.5f,  0.5f,  0.0f,  1.0f,  0.0f,
+            -0.5f,  0.5f, -0.5f,  0.0f,  1.0f,  0.0f
+        };
+
+        private static readonly uint[] Indices =
+        {
+            0, 1, 3,
+            1, 2, 3
+        };
+
+        private static void Main(string[] args)
+        {
+            var options = WindowOptions.Default;
+            options.Size = new Size(800, 600);
+            options.Title = "LearnOpenGL with Silk.NET";
+            window = Window.Create(options);
+
+            window.Load += OnLoad;
+            window.Update += OnUpdate;
+            window.Render += OnRender;
+            window.Closing += OnClose;
+
+            window.Run();
+        }
+
+        private static void OnLoad()
+        {
+            IInputContext input = window.CreateInput();
+            primaryKeyboard = input.Keyboards.FirstOrDefault();
+            if (primaryKeyboard != null)
+            {
+                primaryKeyboard.KeyDown += KeyDown;
+            }
+            for (int i = 0; i < input.Mice.Count; i++)
+            {
+                input.Mice[i].MouseMove += OnMouseMove;
+                input.Mice[i].Scroll += OnMouseWheel;
+            }
+
+            Gl = GL.GetApi(window);
+
+            Ebo = new BufferObject<uint>(Gl, Indices, BufferTargetARB.ElementArrayBuffer);
+            Vbo = new BufferObject<float>(Gl, Vertices, BufferTargetARB.ArrayBuffer);
+            VaoCube = new VertexArrayObject<float, uint>(Gl, Vbo, Ebo);
+
+            VaoCube.VertexAttributePointer(0, 3, VertexAttribPointerType.Float, 6, 0);
+            VaoCube.VertexAttributePointer(1, 3, VertexAttribPointerType.Float, 6, 3);
+
+            //The lighting shader will give our main cube its colour multiplied by the lights intensity
+            LightingShader = new Shader(Gl, "shader.vert", "lighting.frag");
+            //The Lamp shader uses a fragment shader that just colours it solid white so that we know it is the light source
+            LampShader = new Shader(Gl, "shader.vert", "shader.frag");
+
+            //Start a camera at position 3 on the Z axis, looking at position -1 on the Z axis
+            Camera = new Camera(Vector3.UnitZ * 6, Vector3.UnitZ * -1, Vector3.UnitY, Width / Height);
+        }
+
+        private static unsafe void OnUpdate(double deltaTime)
+        {
+            var moveSpeed = 2.5f * (float)deltaTime;
+
+            if (primaryKeyboard.IsKeyPressed(Key.W))
+            {
+                //Move forwards
```

Comment:
nit: imo these comments are a bit unnecessary

Reply:
It's probably best to keep them in though as this is a tutorial after all idk

Reply:
Doesn't hurt